libs/area_reporting.py

```diff
@@ -0,0 +1,72 @@
+import os
+import time
+import logging
+import csv
+from datetime import date
+from collections import deque
+from .utils.mailing import MailService
+from .notifications.slack_notifications import SlackService
+
+logger = logging.getLogger(__name__)
+
+
+class AreaReporting:
+
+    def __init__(self, config, area):
+        self.processing_alerts = False
+        self.config = config
+        self.area = area
+
+        self.occupancy_sleep_time_interval = float(self.config.get_section_dict("App")["OccupancyAlertsTimeout"])
+        self.log_dir = self.config.get_section_dict("Logger")["LogDirectory"]
+        self.idle_time = float(self.config.get_section_dict('Logger')['TimeInterval']) + 0.1
+        self.area_id = self.area['id']
+        self.area_name = self.area['name']
+        self.occupancy_threshold = self.area['occupancy_threshold']
+        self.emails = self.area['emails']
+        self.should_send_email_notifications = self.area['should_send_email_notifications']
+        self.should_send_slack_notifications = self.area['should_send_slack_notifications']
+        self.cameras = [camera for camera in self.config.get_video_sources() if camera['id'] in self.area['cameras']]
+        for camera in self.cameras:
+            camera['file_path'] = os.path.join(self.log_dir, camera['id'], "objects_log")
+            camera['last_processed_time'] = time.time()
+
+        self.mail_service = MailService(config)
+        self.slack_service = SlackService(config)
+
+    def process_area(self):
+        # Sleep for a while so cameras start processing
```

Comment:
What happens if the area doesn't have the notification enabled?.
The thread will process the CSVs without any purpose.

Maybe we can add the sentence 
```
if not (self.should_send_email_notifications or self.should_send_slack_notifications):
    return
```

Reply:
I'll add this condition to https://github.com/neuralet/smart-social-distancing/pull/70/files#diff-2c0d82fa7dc706795dbef4565a49715e6de108f3c09c7c12b4f30d165b62ad45R138 so the thread isn't created in that case